### ISSUE TITLE
IOU Actions - Notifications don't come if subscribed from combined report

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#39706]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.60-5
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/cases/view/2730113
 https://github.com/Expensify/App/issues/39490
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Preconditions: two users A and B.
1. User A made Money Request to user B.
2. User B open IOU report and send a message (this will subscribe the user to notifications).
3. User B is navigated away from the conversation.
4. User A edit Money request: Amount, Merchant, Description or Date.

## Expected Result:
User B receives a notification for the edit action. The notification shows a preview of the action that was made by user A.

## Actual Result:
User B not receives a notification for the edit action.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/4850f9c4-73fb-4180-8aa9-dcfd00ebd01b

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
@jliexpensify I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
Production 


https://github.com/Expensify/App/assets/115492554/6127841e-32d2-46b0-bb6b-aaec2eba6627

**Comment by Julesssss** [comment]
The [checklist](https://github.com/Expensify/App/issues/39588) is small and I see no change related to editing expenses or notifications. I wonder if it might be [this](https://github.com/Expensify/Auth/pull/10418/files) expected change.

**Comment by mountiny** [comment]
I do not think this has to be a blocker.

The reason for this is that the single expense view is rendering the expense report id and the transaction thread report actions are then composed on it.

when we queue the notification for the transaction thread message/ change its not going to show up on the expense report view basically.

@NikkiWines not sure what would be the best way to handle this

**Comment by mountiny** [comment]
Its coming from this PR https://github.com/Expensify/App/pull/36934

**Comment by NikkiWines** [comment]
Yeah agreed, it's not a blocker. Definitely something to investigate further though

**Comment by trjExpensify** [comment]
Ah yeah, this is nice catch! With the [actionable system messages](https://docs.google.com/document/d/1pNWNUG2bF-Q0aXQ9eMdpxvdvuGOkBLjBzJRMAkeN-hA/edit) project underway, we're going to "silence" all system messages by default, so these "edit" expense sys messages aren't going to notify or mark the chat as unread -- so do we need to solve this? 🤔 

CC: @dylanexpensify @deetergp

**Comment by mountiny** [comment]
> we're going to "silence" all system messages by default, so these "edit" expense sys messages aren't going to notify or mark the chat as unread -- so do we need to solve this? 🤔

With the Search page project though, users/ admins can see the individual expenses in the All page of Search and then comment on the expense level/ transaction thread. So there will be a case of non-system message as well

Those comments should show up dynamically same as if you are chatting with someone in Home tab/ expense report

**Comment by Julesssss** [comment]
Not overdue, discussion in progress

**Comment by dylanexpensify** [comment]
From the standpoint of system messages, this won't factor in (since they'll be silent), though I do think Vit brings up a good point about Search page

**Comment by trjExpensify** [comment]
>With the Search page project though, users/ admins can see the individual expenses in the All page of Search and then comment on the expense level/ transaction thread. So there will be a case of non-system message as well

I don't get the concern. If it's a one expense report they would see the combined view in the RHP all the same wouldn't they? :/

**Comment by mountiny** [comment]
>  If it's a one expense report they would see the combined view in the RHP all the same wouldn't they?

Why would they? If its the All expense view and we list only transactions, we should show the transaction thread. Otherwise, if the expense is on multi-expense report we would change to transaction thread. This would be changing based on where the expense is/ context of the report and I think that would be confusing to the user

**Comment by Julesssss** [comment]
Does someone with more context want to take this issue? Based on the above comments I think you'll be able to make a better decision regarding the solution here.

**Comment by trjExpensify** [comment]
>Why would they? If its the All expense view and we list only transactions, we should show the transaction thread.

Because in a "one expense report" case to the user there isn't an additional transaction thread. Even in the search v1 doc, you can see action buttons on one expense report rows in All. 

>Otherwise, if the expense is on multi-expense report we would change to transaction thread. This would be changing based on where the expense is/ context of the report and I think that would be confusing to the user

If that's a real concern, then it would be in the same vein as clicking a preview component in a chat that leads to a one expense report view or a multi-expense report view. The view you see when clicking the preview changes depending on how many expenses there are.

**Comment by mountiny** [comment]
> Because in a "one expense report" case to the user there isn't an additional transaction thread. Even in the search v1 doc, you can see action buttons on one expense report rows in All.

So in search v1, clicking expense that is on single expense report will open the expense report and if we open expense which is on multiple expense report, we will open transaction thread in RHP?  I feel like this will be root cause of lots of bugs and confusion not only for users but also for developers, especially when we add held requests / deleting requests to the game and we would have to switch between what report to show dynamically

**Comment by JmillsExpensify** [comment]
I can see where you're coming from, though this is our design with one-expense reports vs grouped reports. I think we should lean into it for now and we can re-assess based on feedback.

**Comment by trjExpensify** [comment]
Yeah, I don't see how that concern really is any different to a chat view where you click the report preview and either get the combined view or not.

**Comment by mountiny** [comment]
I would love to be proven wrong, but I am happy to leave this up for later once we can get the feel of this in App and get some customer feedback 👍

**Comment by jliexpensify** [comment]
Hello @trjExpensify - apologies if I have misunderstood, but to summarise the discussion: we'll do nothing and close this issue (based off [this](https://github.com/Expensify/App/issues/39706#issuecomment-2040355241) and [this](https://github.com/Expensify/App/issues/39706#issuecomment-2045957380))?

**Comment by trjExpensify** [comment]
IMO, we don't need to solve for it. If a user adds a comment and someone replies to that comment, they will be notified. The system messages are in the process of being silenced anyway. 👍

**Comment by jliexpensify** [comment]
Moved to #Collect (as that's where the Search project is under) and popped on HOLD. Please feel free to close if you disagree!

**Comment by Julesssss** [comment]
> If a user adds a comment and someone replies to that comment

I'm with Tom here that we can close this issue. We're about to rework notifications. If anyone disagree we can re-open and discuss, but I think this is would end up being held for a long time just to become irrelevant.
